torch/csrc/cuda/Module.cpp

```diff
@@ -633,6 +635,7 @@ PyObject* THCPModule_memorySnapshot(PyObject* _unused, PyObject* noargs) {
     segmentDict[total_size_s] = segmentInfo.total_size;
     segmentDict[allocated_size_s] = segmentInfo.allocated_size;
     segmentDict[active_size_s] = segmentInfo.active_size;
+    segmentDict[requested_size_s] = segmentInfo.requested_size;
```

Comment:
Exporting BlockInfo `requested_size` is missing here and in `memory_snapshot.cpp` I also couldn't find a test for the blockinfo/segmentinfo information.

Reply:
Addressed - exporting BlockInfo `requested_size` and added test as part of _check_memory_stat_consistency